### PR TITLE
contracts-bedrock: use `MemorySafeCall`

### PR DESCRIPTION
**Pull request by tynes**
**Description**

Replace `ExcessivelySafeCall` with a local `SafeCall`
that does not copy any calldata back. We do not care about
returndata being returned so this prevents returndata bombs.

### CODE REVIEW

**Comment by tynes** [comment]
Hoping that https://github.com/ethereum-optimism/optimism/pull/3199 fixes the issues we are seeing in CI when it hangs sometimes

File: packages/contracts-bedrock/contracts/universal/StandardBridge.sol [hunk 1]
```diff
@@ -284,7 +285,7 @@ abstract contract StandardBridge is Initializable {
         require(_to != address(this), "StandardBridge: cannot send to self");
 
         emit ETHBridgeFinalized(_from, _to, _amount, _extraData);
-        (bool success, ) = _to.call{ value: _amount }(new bytes(0));
+        bool success = SafeCall.call(_to, gasleft(), _amount, hex"");
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/universal/StandardBridge.sol:288]: Is there a good reason we didn't have ESC here before?

Comment (tynes) [reply packages/contracts-bedrock/contracts/universal/StandardBridge.sol:288]: No good reason, I feel like it should be here

**Comment by mslipper** [comment]
Is this still necessary? I think we've since fixed the SDK, so we don't need to use SafeCall instead anymore.

File: packages/contracts-bedrock/contracts/test/SafeCall.t.sol [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        // don't call the console
+        vm.assume(
+            to != address(0x000000000000000000636F6e736F6c652e6c6f67)
+        );
+        // don't call the create2 deployer
+        vm.assume(
+            to != address(0x4e59b44847b379578588920cA78FbF26c0B4956C)
+        );
+        // don't send funds to self
+        vm.assume(from != to);
```
Comment (maurelian) [inline packages/contracts-bedrock/contracts/test/SafeCall.t.sol:31]: Does something actually break if without these?

**Comment by maurelian** [comment]
> Is this still necessary? I think we've since fixed the SDK, so we don't need to use SafeCall instead anymore.

@mslipper I talked to @tynes and he convinced me it's still beneficial.
@tynes: Can we rename it to `MemorySafeCall`? The `Safe` prefix has been overused and lacks specificity.

Otherwise I'm good with the change.

**Comment by tynes** [comment]
I've rebased this and then updated the name to `MemorySafeCall`

File: packages/contracts-bedrock/contracts/test/SafeCall.t.sol (hunk 2 above)
Comment (tynes) [reply packages/contracts-bedrock/contracts/test/SafeCall.t.sol]: Yes, things revert in unexpected ways when you call the special addresses and the balance assertions fail if you send funds to yourself

**Review by smartcontracts — APPROVED** [review]

**Comment by tynes** [comment]
@smartcontracts Just rebased and fixed the merge conflict

**Review by maurelian — APPROVED** [review]

File: packages/contracts-bedrock/package.json [hunk 3]
```diff
@@ -55,6 +53,7 @@
     "@foundry-rs/easy-foundryup": "^0.1.3",
     "@nomiclabs/hardhat-ethers": "^2.0.0",
     "@nomiclabs/hardhat-waffle": "^2.0.0",
+    "@rari-capital/solmate": "https://github.com/rari-capital/solmate.git#8f9b23f8838670afda0fd8983f2c41e8037ae6bc",
```
Comment (tynes) [inline packages/contracts-bedrock/package.json:56]: Moves solmate to devdeps bc its at a commit hash right now

File: op-bindings/bindings/weth9.go [hunk 4]
```diff
@@ -31,7 +31,7 @@ var (
...
 }
```
Comment (tynes) [inline op-bindings/bindings/weth9.go:35]: The old weth9 has:
- bed565cbea90624f5b56e2f56c4d5fc05493e40220bba9bd15ace6aa4cabb51c
The new weth9 has:
- 2282f823392ab6b42c17b31235784e88e108cd8d10e77cd732aa3e1e7a88a7d

This looks like a bytecode hash

The diff is right at the very end

**Pull request merged**